tests/subsys/storage/flash_map/src/main.c

```diff
@@ -32,51 +32,95 @@ ZTEST(flash_map, test_flash_area_disabled_device)
 	zassert_equal(rc, -ENODEV, "Open did not fail");
 }
 
+ZTEST(flash_map, test_flash_area_device_is_ready)
+{
+	const struct flash_area no_dev {
+		.fa_dev = NULL;
+	};
+
+	zassert_false(flash_area_device_is_ready(NULL));
+	zassert_false(flash_area_device_is_ready(no_dev));
```

Comment:
There appears to be something wrong here.

Reply:
yes, and merged.